codec.go

```diff
@@ -272,9 +262,9 @@ func (c *Codec) encodePayload(ctx context.Context, payload *common.Payload) (*co
 		return nil, fmt.Errorf("unable to unmarshal put response: %w", err)
 	}
 
-	result, err := converter.GetDefaultDataConverter().ToPayload(remotePayload{
+	result, err := converter.GetDefaultDataConverter().ToPayload(codecpb.RemotePayload{
```

Comment:
What do we gain here really by using a `codecpb.RemotePayload`? This is not something which goes over the wire or gets used by gRPC. The default remote codec does in the end use `json.Marshal` (since it needs to be able to encode any payload really). See https://github.com/temporalio/sdk-go/blob/master/converter/codec.go#L456 So looking at the generated struct, the thing which matters are the `json` annotations. Everything else is pretty much ignored

```
type RemotePayload struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	// The content of the original payload's metadata
	Metadata map[string][]byte `protobuf:"bytes,1,rep,name=metadata,proto3" json:"metadata,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
	// Number of bytes in the original payload's data
	Size uint64 `protobuf:"varint,2,opt,name=size,proto3" json:"size,omitempty"`
	// Digest of the payload data, prefixed with the algorithm, e.g. sha256:deadbeef
	Digest string `protobuf:"bytes,3,opt,name=digest,proto3" json:"digest,omitempty"`
	// The key to retrieve the payload from remote storage
	Key string `protobuf:"bytes,4,opt,name=key,proto3" json:"key,omitempty"`
}
```

I guess something similar is true for the Python code.